webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/rudder/rest/lift/HookApi.scala

```diff
@@ -0,0 +1,76 @@
+package com.normation.rudder.rest.lift
+
+import better.files.File
+import com.normation.errors.IOResult
+import com.normation.rudder.api.ApiVersion
+import com.normation.rudder.rest.ApiPath
+import com.normation.rudder.rest.AuthzToken
+import com.normation.rudder.rest.EndpointSchema
+import com.normation.rudder.rest.RestExtractorService
+import com.normation.rudder.rest.RestUtils
+import net.liftweb.http.LiftResponse
+import net.liftweb.http.Req
+import com.normation.rudder.rest.{HookApi => API}
+import net.liftweb.common.EmptyBox
+import net.liftweb.common.Full
+import net.liftweb.json.JsonAST.JArray
+import net.liftweb.json.JsonAST.JField
+import net.liftweb.json.JsonDSL._
+
+import com.normation.box._
+
+
+class HookApi(
+    restExtractorService  : RestExtractorService
+  , apiVService           : HookApiService
+) extends LiftApiModuleProvider[API] {
+
+  def schemas = API
+
+  override def getLiftEndpoints(): List[LiftApiModule] = {
+
+    API.endpoints.map(e => e match {
+      case API.GetHooks                      => GetHooks
+    })
+  }
+
+  object GetHooks extends LiftApiModule0 {
+    val schema = API.GetHooks
+    val restExtractor = restExtractorService
+
+    def process0(version: ApiVersion, path: ApiPath, req: Req, params: DefaultParams, authzToken: AuthzToken): LiftResponse = {
+      apiVService.getHooks(schema, params)
+    }
+  }
+}
+
+class HookApiService {
+
+  def getHooks(schema: EndpointSchema, params: DefaultParams): LiftResponse  = {
+    implicit val action = schema.name
+    implicit val prettify = params.prettify
+
+    val result = for {
+      path   <- IOResult.effect(File("/opt/rudder/etc/hooks.d"))
+      hooks  <- IOResult.effect {
+        path.list.filter(_.isDirectory).map { d =>
+          ( d.name
+            , d.children.map(_.name).filter(h => h != "readme.adoc" && h != "hook-template").toList
+          )
+        }.toList.sortBy(_._1)
+      }
+    } yield {
+      ( ("root"   -> path.toString)
```

Comment:
Don't define directly the answer, you should do like Rule api and define a JRHooksList  object (JR = JSON Response ) in webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/apidata/JsonResponseObjects.scala 

You will need to define related encodes in RudderJsonEncoders 

The goal here is to use zio.json instead of lift-json to generate json response